include/json/reader.h

```diff
@@ -262,7 +268,22 @@ class JSON_API CharReader {
    * error occurred.
    */
   virtual bool parse(char const* beginDoc, char const* endDoc, Value* root,
-                     String* errs) = 0;
+                     String* errs);
+
+  /** \brief Returns a vector of structured errors encountered while parsing.
+   * Each parse call resets the stored list of errors. Each call to this
+   * function reads the stored list of errors and builds a vector of
+   * structured errors to return.
+  */
+  std::vector<StructuredError> getStructuredErrors() const;
+
+  class Impl {
```

Comment:
I think this should be `protected` API, not `public`.

Derived classes need to know what an `Impl` is but nobody else does.